### PR TITLE
Heat Map Colors

### PR DESCRIPTION
**Pull request by JasonLin43212**
Added colors to the heat map according to data values.
You have the option to select a year and a score type to see the different values.

### CODE REVIEW

File: frontend/src/heat_map/democracy_viz.js [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+        )
+    }
+}
+DemocracyMap.propTypes = {
+    democracyData: PropTypes.array,
+    scoreType: PropTypes.string,
+    year: PropTypes.string,
+};
+
+export class MapPath extends React.Component {
```
Comment (ryaanahmed) [inline frontend/src/heat_map/democracy_viz.js]: This looks like copy/paste from the map quiz -- can you factor out? We'll need this all over the place. Please coordinate with the map quiz team @meesuekim @maoivy @bengao01 to make sure you're not making any contradictory changes to this component.

File: frontend/src/heat_map/democracy_viz.js [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+            const map_data = this.project_features_and_create_svg_paths(geo_json);
+            this.setState({
+                map_data: map_data,
+            });
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    project_features_and_create_svg_paths(geo_json) {
```
Comment (ryaanahmed) [inline frontend/src/heat_map/democracy_viz.js]: also copy/paste from map quiz -- please factor this common mapping functionality out.

File: frontend/src/heat_map/democracy_viz.js [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+        "min": 0,
+        "max": 1
+    },
+};
+
+export class DemocracyMap extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            lng: 29,
```
Comment (ryaanahmed) [inline frontend/src/heat_map/democracy_viz.js]: there's some cruft leftover here from map quiz that we can remove now, so we don't let it build up!

File: frontend/src/heat_map/democracy_viz.js [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+                    democracyData={this.state.democracyData}
+                    scoreType={this.state.scoreType}
+                    year={this.state.year}
+                />
+            </>
+        )
+    }
+}
+
+const scoreTypeRanges = {
```
Comment (ryaanahmed) [inline frontend/src/heat_map/democracy_viz.js]: for neatness, please put constants at the top of the file -- this particular constant is fine for prototyping purposes, but let's maybe move this to a data file and read it in the backend. we're going to want it all over the place.

File: frontend/src/heat_map/democracy_viz.js [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+            const democracy_data = await res.json();
+            this.setState({
+                democracyData: JSON.parse(democracy_data),
+            });
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    handleScoreTypeChange (e) {
```
Comment (ryaanahmed) [inline frontend/src/heat_map/democracy_viz.js]: nitpick -- please no space before the opening paren here

File: backend/app/views.py [hunk 6]
```diff
@@ -22,10 +23,54 @@ def load_africa_geojson() -> dict:
...
+    and return a parsed json array
+    """
+    filename = 'lieberman_afr_data.csv'
+    path = Path(settings.BACKEND_DATA_DIR, filename)
+    democracy_data = []
+    with open(path, encoding='utf-8') as democracy_score_file:
+        reader = csv.reader(democracy_score_file, delimiter=',')
+        headers = []
+        current_country_data = {"democracy_scores": {}}
+        for row in reader:
```
Comment (ryaanahmed) [inline backend/app/views.py:59]: A little too much magic number stuff happening in this file read. Let's discuss offline.

**Review by ryaanahmed — CHANGES_REQUESTED** [review]
REALLY GOOD so far!

I've made some comments inline for cleanup.

Also: we're going to need to test a bunch of stuff here, but while we're in this rapid prototyping / things changing quickly mode, we can lay off. Once the general architecture stabilizes by the beginning of next week, we'll need to get rigorous about that.

**Comment by JasonLin43212** [comment]
@ryaanahmed finished making changes. Also added in normalizing function.

**Review by ryaanahmed — APPROVED** [review]
looks good! time to test test test all of this.

re: normalization -- thanks for that; I suspect for some of these metrics we're not going to want to normalize, so let's make sure to flag that up to Evan at some point.

**Pull request merged**